src/crypto/local.js

```diff
@@ -287,6 +293,74 @@ function validResponseCompression(compression, setting) {
     return true;
 }
 
+// Commitments verification
+
+/**
+ * Parse a DER-encoded signature.
+ * @param {string} derSignature - A signature in DER format.
+ * @return {sjcl.bitArray} a signature object for sjcl library.
+ */
+function parseSignaturefromDER(derSignature) {
+    try {
+        const sigBN = parseKeys.signature.decode(
+            Buffer.from(derSignature, "base64"),
+            "der"
+        );
+        return sjcl.codec.bytes.toBits(
+            sjcl.bitArray.concat(sigBN.r.toArray(), sigBN.s.toArray())
+        );
+    } catch (error) {
+        throw new Error(
+            "[privacy-pass]: Failed on parsing commitment signature. " + error
+        );
+    }
+}
+
+/**
+ * Parse a DER-encoded publick key.
+ * @param {string} derPublicKey - A public key in DER format.
+ * @return {sjcl.ecc.ecdsa.publicKey} a public key for sjcl library.
+ */
+function parsePublicKeyfromDER(derPublicKey) {
+    const OIDCurves = [];
+    OIDCurves["c256"] = "1.2.840.10045.3.1.7"; // oid(prime256v1)
+
+    try {
+        const pkJson = parseKeys(derPublicKey);
+        assert.equal(pkJson.type, "ec");
+        assert.equal(
+            pkJson.data.algorithm.curve.join("."),
+            OIDCurves[sjcl.ecc.curveName(CURVE)]
+        );
+        const point = sec1DecodeFromBytes(pkJson.data.subjectPublicKey.data);
+        return new sjcl.ecc.ecdsa.publicKey(CURVE, point);
+    } catch (error) {
+        throw new Error(
+            "[privacy-pass]: Failed on parsing public key. " + error
+        );
+    }
+}
+
+/**
+ * Verify the signature of commitments.
+ * @param {json} comms - commitments to verify
+ * @param {string} derPublicKey - A public key in DER format.
+ * @return {boolean} True, if the commitment signature is valid.
+ */
+function verifyCommitments(comms, derPublicKey) {
```

Comment:
I think we should also verify the expiry date of the commitment in here. I couldn't seem to find it?

Reply:
I would handle this verification in another PR.